monitor/api/models/models.py

```diff
@@ -45,51 +56,65 @@ def get_test_coverages(self) -> Dict[str, ModelCoverageSchema]:
         return coverages
 
     @staticmethod
-    def _normalize_dbt_model_dict(model: ModelSchema, is_source: bool = False) -> NormalizedModelSchema:
-        model_name = model.name
+    def _normalize_dbt_artifact_dict(
+        artifact: Union[ModelSchema, ExposureSchema, SourceSchema],
+        type: Literal["model", "source", "exposure"]
+    ) -> Union[NormalizedExposureSchema, NormalizedModelSchema, NormalizedSourceSchema]:
+        artifact_name = artifact.name
 
-        owners = model.owners
+        owners = artifact.owners
         if owners:
             loaded_owners = try_load_json(owners)
             if loaded_owners is not None:
                 owners = loaded_owners
             else:
                 owners = [owners]
 
-        tags = model.tags
+        tags = artifact.tags
         if tags:
             loaded_tags = try_load_json(tags)
             if loaded_tags is not None:
                 tags = loaded_tags
             else:
                 tags = [tags]
         
-        normalized_model = json.loads(model.json())
-        normalized_model['owners'] = owners
-        normalized_model['tags'] = tags
-        normalized_model['model_name'] = model_name
-        normalized_model['normalized_full_path'] = ModelsAPI._normalize_model_path(
-            model_path=model.full_path,
-            model_package_name=model.package_name,
-            is_source=is_source
+        normalized_artifact = json.loads(artifact.json())
+        normalized_artifact['owners'] = owners
+        normalized_artifact['tags'] = tags
+        normalized_artifact['model_name'] = artifact_name
+        normalized_artifact['normalized_full_path'] = ModelsAPI._normalize_artifact_path(
+            artifact_path=artifact.full_path,
+            artifact_package_name=artifact.package_name,
+            type=type
         )
-        return NormalizedModelSchema(**normalized_model)
+
+        if type == "exposure":
+            return NormalizedExposureSchema(**normalized_artifact)
+        elif type == "model":
+            return NormalizedModelSchema(**normalized_artifact)
+        elif type == "source":
```

Comment:
What happens if the type is not one of them? what do you return in that case? also, I think Elon adds a new type of snapshot to dbt_models, let's verify with him?

Reply:
Yeah but I can't change my PR before he merges his then.
Plus do we show snapshot as models in the sidebar/tables?
If not what is the case here?
I think that those question should be handled in Elon's PR.
If we change the models table in a PR, we need to change everything that is relied on it in that PR (models query, etc)